### PR TITLE
first successfull implementation of the diff

### PR DESCRIPTION
**Pull request by AntonioSouls**
I've finally modified diff.py code and it works successfully. I'm waiting for next task!! Please check that everything works

### CODE REVIEW

File: memory_diff/diff.py [hunk 1]
```diff
@@ -3,46 +3,38 @@
...
-    
-    def __init__(self,file_new: str, file_old:str) -> None: 
+    # CLASS CONSTRUCTOR: the object takes three files as parameters, where the first one we insert is the "new" one, the second one is the "old" one, the third one is the file in which 
+    # we save the differences
+    def __init__(self,file_new: str, file_old:str, file_diff:str) -> None: 
         self.file_new = file_new
         self.file_old = file_old
+        self.file_diff = file_diff
         self.lista_file_new = None
         self.lista_file_old = None
```
Comment (samirsalman) [inline memory_diff/diff.py]: use english name for the variables

File: memory_diff/diff.py [hunk 2]
```diff
@@ -3,46 +3,38 @@
...
-        # #             if x[0] == '-':
-        # #                 self.lista_file_old.insert(diff_result.index(x), self.lista_file_new[diff_result.index(x)])
-        # #             else:
-        # #                 self.lista_file_old.pop(diff_result.index(x))
-        # #                 self.lista_file_old.insert(diff_result.index(x), self.lista_file_new[diff_result.index(x)])
+    # Doing diff between two files: an Old File and a New File and saving into a third file only the differences between the preavious files
+    def diff_function(self):
+        file_object_new = open(self.file_new, 'r')                             # Opening the new file for reading
+        file_object_old = open(self.file_old, 'r')                             # Opening the old file for reading
+        self.lista_file_new = file_object_new.readlines()                      # Saving the new file's lines into a list
```
Comment (samirsalman) [inline memory_diff/diff.py]: same as before

File: memory_diff/diff.py [hunk 3]
```diff
@@ -3,46 +3,38 @@
...
-        # #             self.lista_file_old.pop(diff_result.index(x))
-        # #         else:
-        # #             if x[0] == '-':
-        # #                 self.lista_file_old.insert(diff_result.index(x), self.lista_file_new[diff_result.index(x)])
-        # #             else:
-        # #                 self.lista_file_old.pop(diff_result.index(x))
-        # #                 self.lista_file_old.insert(diff_result.index(x), self.lista_file_new[diff_result.index(x)])
+    # Doing diff between two files: an Old File and a New File and saving into a third file only the differences between the preavious files
+    def diff_function(self):
+        file_object_new = open(self.file_new, 'r')                             # Opening the new file for reading
```
Comment (samirsalman) [inline memory_diff/diff.py]: in python you can use the `with open(FILENAME, MODE) as f_name: ...` (https://stackoverflow.com/questions/9282967/how-to-open-a-file-using-the-open-with-statement)

File: memory_diff/diff.py [hunk 4]
```diff
@@ -3,46 +3,38 @@
...
+        file_object_old.close()                                                # Closing the old file
         
-        # object_file_old = open(self.file_old, 'w')                            #apro il vecchio file per riscriverci dentro la lista modificata
-        # object_file_old.writelines(self.lista_file_old)                       #La scrivo
-        # object_file_old.close()                                                #chiudo il file e lo restituisco
-        # return self.file_old  
-        return                                                  
\ No newline at end of file
+                                                                              # Starting the research of the differences between two files
+        for stringa_new in self.lista_file_new:
```
Comment (samirsalman) [inline memory_diff/diff.py]: use english names

File: memory_diff/diff.py [hunk 5]
```diff
@@ -3,46 +3,38 @@
...
+            esiste: bool = False
```
Comment (samirsalman) [inline memory_diff/diff.py]: same

File: memory_diff/diff.py [hunk 6]
```diff
@@ -3,46 +3,38 @@
...
+            i:int = 0
+            while i<len(self.lista_file_old) and esiste == False:
+                if stringa_new == self.lista_file_old[i] :
+                    esiste = True
+                i=i+1
+            if(esiste==False):
```
Comment (samirsalman) [inline memory_diff/diff.py]: can be -> `if not esiste:` or `if not exist:` after the renaming

File: memory_diff/diff.py [hunk 7]
```diff
@@ -3,46 +3,38 @@
...
+                self.lista_file_diff.append(stringa_new)
+
+                                                                            # Saving the differences into a third file called diff_file
+        file_object_diff = open(self.file_diff, 'w')
+        file_object_diff.writelines(self.lista_file_diff)
+        file_object_diff.close()
+        return
```
Comment (samirsalman) [inline memory_diff/diff.py]: remove the return if you're returning `None`

File: test_memory_diff/Matecat_New.tmx [hunk 8]
```diff
@@ -6,10 +6,10 @@
       <prop type="matecat-tee">41698</prop>
       <prop type="matecat-match">MT</prop>
       <tuv xml:lang="en-GB">
-        <seg>Important legal notice, really important</seg>
+	  <seg>Importante avviso legale, veramente importante</seg>
```
Comment (samirsalman) [inline test_memory_diff/Matecat_New.tmx]: remove the tmx files from github and add them to the .gitignore, both old, new and diff file

File: test_memory_diff/test_diff.py [hunk 9]
```diff
@@ -5,8 +5,9 @@
 class Test_TestIncrementDecrement(unittest.TestCase):
 
     def test_saving_diff(self):
-        diff_object = Diff('/mnt/c/Users/Antonio/Desktop/memory-diff/test_memory_diff/fileNew.txt', '/mnt/c/Users/Antonio/Desktop/memory-diff/test_memory_diff/fileOld.txt')
-        modified_file = diff_object.saving_diff()
+        diff_object = Diff('/mnt/c/Users/Antonio/Desktop/memory-diff/test_memory_diff/Matecat_New.tmx', '/mnt/c/Users/Antonio/Desktop/memory-diff/test_memory_diff/Matecat_Old.tmx','/mnt/c/Users/Antonio/Desktop/memory-diff/test_memory_diff/diff_file.tmx')
```
Comment (samirsalman) [inline test_memory_diff/test_diff.py]: put the path as input of the test function, is not a best practice to hardcode paths

Comment (AntonioSouls) [reply test_memory_diff/test_diff.py:8]: I don't understand what i've to do

Comment (samirsalman) [reply test_memory_diff/test_diff.py]: instead of put the file path inside the code you should take the file path as input

**Review by samirsalman — APPROVED** [review]

**Pull request merged**